src/interp/interp.cc

```diff
@@ -260,31 +268,46 @@ void Store::Collect() {
     thread->Mark();
   }
 
-  // TODO: better GC algo.
-  // Loop through all newly marked objects and mark their referents.
-  std::vector<bool> all_marked(object_count, false);
-  bool new_marked;
-  do {
-    new_marked = false;
-    for (size_t i = 0; i < object_count; ++i) {
-      if (!all_marked[i] && marks_[i]) {
-        all_marked[i] = true;
-        objects_.Get(i)->Mark(*this);
-        new_marked = true;
-      }
-    }
-  } while (new_marked);
+  // This vector is often empty since the default maximum
+  // recursion is usually enough to mark all objects.
```

Comment:
Do we even have test cases that trigger this condition?  If we don't I guess we should construct at least one?

Reply:
Good suggestion. What is the easiest way to construct such a chain? Tables, which contain a single element to another Table? If I need to do it in C++ level, where should I put the test?

Thank you for the review!

Reply:
I'm not sure I'm afraid, I'm not to familiar with the the GC/interpreter yet.

I think `InterpGCTest` in `src/test-interp.cc` is probably a good place to start.

Reply:
You are right. I have added a test. But before other changes, I will wait your reply what should we do with GCContext.

Reply:
I recommend defining GCContext as you have, but including an instance in the Store rather than a pointer.  That way you get (a) fewer allocations and (b) no lifetime issues with pointers to stack data.